### ISSUE TITLE
[PATCH] POD-only mention sv_setsv does get magic but not set magic

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#13405]
Migrated from [rt.perl.org#120490](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=120490) (status was 'resolved')

Searchable as RT120490$

**Comment by p5pRT** [comment]
### From @bulk88

#### Created by @bulk88



See attached patch\.

<details><summary>Perl Info</summary>

```

Flags:
    category=docs
    severity=low

Site configuration information for perl 5.19.6:

Configured by Owner at Thu Oct 24 19:59:31 2013.

Summary of my perl5 (revision 5 version 19 subversion 6) configuration:
  Derived from:
  Platform:
    osname=MSWin32, osvers=5.1, archname=MSWin32-x86-multi-thread
    uname=''
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    useithreads=define, usemultiplicity=define
    useperlio=define, d_sfio=undef, uselargefiles=define, usesocks=undef
    use64bitint=undef, use64bitall=undef, uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cl', ccflags ='-nologo -GF -W3 -O1 -MD -Zi -DNDEBUG -GL -DWIN32 
-D_CONSOLE -DNO_STRICT  -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT 
-DPERL_IMPLICIT_SYS -DUSE_PERLIO -D_USE_32BIT_TIME_T',
    optimize='-O1 -MD -Zi -DNDEBUG -GL',
    cppflags='-DWIN32'
    ccversion='13.10.6030', gccversion='', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=undef, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='__int64', 
lseeksize=8
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='link', ldflags ='-nologo -nodefaultlib -debug -opt:ref,icf 
-ltcg  -libpath:"c:\perl519\lib\CORE"  -machine:x86'
    libpth="C:\Program Files\Microsoft Visual Studio .NET 2003\VC7\lib"
    libs=oldnames.lib kernel32.lib user32.lib gdi32.lib winspool.lib  
comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib  
netapi32.lib uuid.lib ws2_32.lib mpr.lib winmm.lib  version.lib 
odbc32.lib odbccp32.lib comctl32.lib msvcrt.lib
    perllibs=oldnames.lib kernel32.lib user32.lib gdi32.lib 
winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib 
oleaut32.lib  netapi32.lib uuid.lib ws2_32.lib mpr.lib winmm.lib  
version.lib odbc32.lib odbccp32.lib comctl32.lib msvcrt.lib
    libc=msvcrt.lib, so=dll, useshrplib=true, libperl=perl519.lib
    gnulibc_version=''
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags='-dll -nologo -nodefaultlib -debug 
-opt:ref,icf -ltcg  -libpath:"c:\perl519\lib\CORE"  -machine:x86'

Locally applied patches:
    uncommitted-changes


@INC for perl 5.19.6:
    C:/perl519/site/lib
    C:/perl519/lib
    .


Environment for perl 5.19.6:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=C:\perl519\bin;C:\Program Files\Microsoft Visual Studio .NET 
2003\Common7\IDE;C:\Program Files\Microsoft Visual Studio .NET 
2003\VC7\BIN;C:\Program Files\Microsoft Visual Studio .NET 
2003\Common7\Tools;C:\Program Files\Microsoft Visual Studio .NET 
2003\Common7\Tools\bin\prerelease;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\system32\wbem;
    PERL_BADLANG (unset)
    SHELL (unset)


```
</details>

**Comment by p5pRT** [comment]
### From @bulk88

<details><summary>0001-POD-only-mention-sv_setsv-does-get-magic-but-not-set.patch</summary>

```diff
From 4db2be0644a9380878a0eb2eab262a09ee7b520f Mon Sep 17 00:00:00 2001
From: Daniel Dragan <bulk88@hotmail.com>
Date: Fri, 8 Nov 2013 05:38:51 -0500
Subject: [PATCH] POD-only mention sv_setsv does get magic but not set magic

---
 sv.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/sv.c b/sv.c
index 8ef01c9..c1ea40d 100644
--- a/sv.c
+++ b/sv.c
@@ -3662,9 +3662,9 @@ Perl_sv_utf8_decode(pTHX_ SV *const sv)
 
 Copies the contents of the source SV C<ssv> into the destination SV
 C<dsv>.  The source SV may be destroyed if it is mortal, so don't use this
-function if the source SV needs to be reused.  Does not handle 'set' magic.
-Loosely speaking, it performs a copy-by-value, obliterating any previous
-content of the destination.
+function if the source SV needs to be reused.  Does not handle 'set' magic on
+destination SV. Calls 'get' magic on source SV. Loosely speaking, it performs a
+copy-by-value, obliterating any previous content of the destination.
 
 You probably want to use one of the assortment of wrappers, such as
 C<SvSetSV>, C<SvSetSV_nosteal>, C<SvSetMagicSV> and
-- 
1.7.9.msysgit.0


```
</details>

**Comment by p5pRT** [comment]
### From @bulk88

Father C\, I see applied my patch as a731eb084d691f568ce9cedc364aa1782d3d85f5 but you short after made commit "Consistent spaces after dots in sv\.c apidocs" a05ea1cf8be01f657bfd7e533d25812d0eeb048c \.

In that commit I see you did


---
@&#8203;@&#8203; \-3663\,7 \+3663\,8 @&#8203;@&#8203; Perl\_sv\_utf8\_decode\(pTHX\_ SV \*const sv\)
 Copies the contents of the source SV C\<ssv> into the destination SV
 C\<dsv>\.  The source SV may be destroyed if it is mortal\, so don't use this
 function if the source SV needs to be reused\.  Does not handle 'set' magic on
\-destination SV\. Calls 'get' magic on source SV\. Loosely speaking\, it performs a
\+destination SV\. C alls 'get' magic on
\+source SV\.  Loosely speaking\, it performs a
 copy\-by\-value\, obliterating any previous content of the destination\.
 
 You probably want to use one of the assortment of wrappers\, such as


---
You probably want to fix the spacing and spelling\.

\-\- 
bulk88 ~ bulk88 at hotmail\.com

**Comment by p5pRT** [comment]
### From perl5-porters@perl.org

Daniel Dragan wrote&#8203;:
> You probably want to fix the spacing and spelling\.

Sorry\, I screwed that up\.  I have now fixed that in e657afc\.

Also\, sorry\, I had not got to stating in this ticket that I had
applied it\.

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
@cpansprout - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @bulk88

On Fri Nov 08 17&#8203;:42&#8203;:57 2013\, perl5\-porters@&#8203;perl\.org wrote&#8203;:
> Daniel Dragan wrote&#8203;:
> > You probably want to fix the spacing and spelling\.
> 
> Sorry\, I screwed that up\.  I have now fixed that in e657afc\.
> 
> Also\, sorry\, I had not got to stating in this ticket that I had
> applied it\.
> 
> 

I'm NOT worried about not finding out if my patch was applied\. Sometimes committers forget to put in the ticket it was applied\. If after 24\-48 hrs the ticket isn't updated I'll just write in the git info myself and forget about the ticket \(eventually jkeenan will come by to close it\)\. It is not a problem\. I watch the blead git stream for my own patches\, and sometimes look at other author's commits\.

\-\- 
bulk88 ~ bulk88 at hotmail\.com
